### PR TITLE
chore(forms): Extended features docs

### PR DESCRIPTION
**Pull request by henit**
Doing the following (inside the Forms-extension):
- Rewrite Extended feature page with sections instead of automatic component lists
- Moving DataContext and Value from "Create your own component" to "Extended features"
- Moving Field from "Create your own component" to a separate menu item above the field components in the left menu
- Remove duplicated list of components between the Field page and the two Base/Feature field pages

### CODE REVIEW

**Comment by tujoworker** [comment]
We should check if there are any affected/existing links/urls we need to change. 

@langz can you check that?

File: packages/dnb-design-system-portal/src/docs/uilib/extensions/forms/fields.mdx [hunk 1]
```diff
@@ -49,13 +41,9 @@ The base components exists of several kinds, such as the:
...
 
-## All base components
-
-<ListBaseFieldComponents />
-
-## Feature fields
+## Feature components
 
-<ListFeatureFieldComponents />
+[Feature fields](/uilib/extensions/forms/feature-fields/) build on top of the base components to add more specific features targeting a given type of data.
```
Comment (langz) [inline packages/dnb-design-system-portal/src/docs/uilib/extensions/forms/fields.mdx:46]:
```suggestion
[Feature fields](/uilib/extensions/forms/feature-fields/) builds on top of the base components to add more specific features targeting a given type of data.
```

**Review by langz — APPROVED** [review]
Thanks for the great effort 👏 
I added a [comment](https://github.com/dnbexperience/eufemia/pull/2978#issuecomment-1845317466) with a few broken links that I could find, seems like they are already fixed 🙇

**Comment by langz** [comment]
> > I can take a quick round of testing after the latest changes is ready in the [preview](https://eufemia-git-chore-forms-extended-features-docs-eufemia.vercel.app/), to see if I can find anything related to the latest changes/commits.
> 
> 
> 
> @langz Have tou re-checked? Don't want to merge before you are finished.
> 
> 

I can re-check in about 30min🚀

**Review by tujoworker — APPROVED** [review]
Looks good. Thank you 👍

**Comment by tujoworker** [comment]
But maybe we should improve the "about fields" titles?

Screenshot 2023-12-08 at 09 51 49

vs

Screenshot 2023-12-08 at 09 52 31

**Comment by langz** [comment]
> But maybe we should improve the "about fields" titles to something like this?
> 
> Screenshot 2023-12-08 at 09 51 49
> vs
> 
> Screenshot 2023-12-08 at 09 52 31

Great idea, I can make this improvement, either as part of this PR or as part of a new PR

File: packages/dnb-design-system-portal/src/docs/uilib/extensions/forms/base-fields/Option/info.mdx [hunk 2]
```diff
@@ -6,4 +6,4 @@ showTabs: true
 
 `Field.Option` is a part for building selection inputs with Field.Select.
 
-There is a corresponding [Value.Option](/uilib/extensions/forms/create-component/Value/Option) component.
+There is a corresponding [Value.Option](/uilib/extensions/forms/extended-features/Value/Option) component.
```
Comment (langz) [inline packages/dnb-design-system-portal/src/docs/uilib/extensions/forms/base-fields/Option/info.mdx:9]:
```suggestion
```

I think we can/should remove this, as the link doesn't work, and it perhaps doesn't exist any Value.Option component? 🤔

**Review by langz — APPROVED** [review]
Thanks again :) 

The links works way better this time around, only broken link I found when re-testing was [comment](https://github.com/dnbexperience/eufemia/pull/2978#pullrequestreview-1771907607) 

I'll add a fix for the  [comment](https://github.com/dnbexperience/eufemia/pull/2978#pullrequestreview-1771907607) and improve upon the suggestion https://github.com/dnbexperience/eufemia/pull/2978#issuecomment-1846789296, and merge this 🙇

File: packages/dnb-design-system-portal/src/docs/uilib/extensions/forms/base-fields/Option/info.mdx (hunk 2 above)
Comment (langz) [reply packages/dnb-design-system-portal/src/docs/uilib/extensions/forms/base-fields/Option/info.mdx:9]: I'll remove this line, as I couldn't find `Value.Option` in packages/dnb-eufemia/src/extensions/forms/Value/index.ts

**Comment by langz** [comment]
I've now [added a commit](https://github.com/dnbexperience/eufemia/pull/2978/commits/9a01bcf5e28e73f2b0787540d25b7aefefb26c78) to improve upon the[ following suggestion](https://github.com/dnbexperience/eufemia/pull/2978#issuecomment-1846789296).

Does it look good @tujoworker?

Here's a [link to the preview ](https://eufemia-git-chore-forms-extended-features-docs-eufemia.vercel.app/uilib/extensions/forms/fields/)

And here's a screenshot:
Screenshot 2023-12-08 at 12 32 14

**Comment by tujoworker** [comment]
@langz yes, you can merge them ✅

**Pull request merged**